interfaces/builtin/socketcan.go

```diff
@@ -0,0 +1,61 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const socketCanSummary = `allows configuration and use of SocketCAN network interfaces`
+
+const socketCanBaseDeclarationSlots = `
+  socketcan:
+    allow-installation:
+      slot-snap-type:
+        - core
+`
+
+const socketCanConnectedPlugAppArmor = `
+# Description: Can access SocketCAN networks.
+network can,
+network netlink raw,
```

Comment:
Can you please provide the rationale for using netlink sockets? A quick google on wikipedia seems to imply this is not required (see: https://en.wikipedia.org/wiki/SocketCAN)